### PR TITLE
Year on year reducer fix

### PR DESCRIPTION
**Pull request by janhenrikoverland**
Ready for review.

### CODE REVIEW

File: packages/app/src/components/Dimensions/DataDimension/DataDimension.js [hunk 1]
```diff
@@ -219,7 +220,7 @@ export class DataDimension extends Component {
                             />
                         </div>
                         <SelectedItems
-                            items={this.props.selectedItems.dx}
```
Comment (edoardo) [inline packages/app/src/components/Dimensions/DataDimension/DataDimension.js]: Should we use `dxId` everywhere?

File: packages/app/src/components/Dimensions/DataDimension/DataDimension.js [hunk 2]
```diff
@@ -236,16 +237,20 @@ export class DataDimension extends Component {
...
     onUpdate: PropTypes.func.isRequired,
 };
 
+DataDimension.defaultProps = {
+    selectedItems: [],
+};
+
 const mapStateToProps = state => ({
-    selectedItems: sGetUiItems(state),
+    selectedItems: sGetUiItems(state).dx,
```
Comment (edoardo) [inline packages/app/src/components/Dimensions/DataDimension/DataDimension.js]: `dxId` ?

File: packages/app/src/reducers/__tests__/current.spec.js [hunk 3]
```diff
@@ -34,36 +40,81 @@ describe('reducer: current', () => {
 
     it('SET_CURRENT_FROM_UI: should set the current from the ui state section', () => {
         const ui = {
-            type: 'COLUMN',
-            layout: { columns: ['dx'], rows: ['ou'], filters: ['pe'] },
+            type: COLUMN,
+            layout: { columns: [dxId], rows: [ouId], filters: [peId] },
+            itemsByDimension: {
+                dx: ['dxItemId1', 'dxItemId2'],
```
Comment (edoardo) [inline packages/app/src/reducers/__tests__/current.spec.js]: `[dxId]` ? and `[ouId]`, `[peId]` below?

File: packages/app/src/reducers/__tests__/current.spec.js [hunk 4]
```diff
@@ -34,36 +40,81 @@ describe('reducer: current', () => {
...
+
+    it('SET_CURRENT_FROM_UI: should set current on a year on year format from the ui state section', () => {
+        const ui = {
+            type: YEAR_ON_YEAR,
+            layout: { columns: [], rows: [], filters: [dxId, ouId] },
             itemsByDimension: {
-                dx: ['dxId1', 'dxId2'],
-                ou: ['ouId1', 'ouId2'],
-                pe: ['peId1'],
+                dx: ['dxItemId1', 'dxItemId2'],
```
Comment (edoardo) [inline packages/app/src/reducers/__tests__/current.spec.js]: `[dxId]`, `[ouId]` ?

File: packages/app/src/reducers/__tests__/ui.spec.js [hunk 5]
```diff
@@ -109,6 +110,27 @@ describe('reducer: ui', () => {
...
     });
 
+    it(`${SET_UI_TYPE}: should set the type, layout and items on the year on year format`, () => {
+        const expectedState = {
+            ...DEFAULT_UI,
+            type: YEAR_ON_YEAR,
+            layout: {
+                columns: [],
+                rows: [],
+                filters: ['ou', 'dx'],
```
Comment (edoardo) [inline packages/app/src/reducers/__tests__/ui.spec.js]: `dxId`, `ouId`?

File: packages/app/src/reducers/__tests__/ui.spec.js [hunk 6]
```diff
@@ -109,6 +110,27 @@ describe('reducer: ui', () => {
...
+            },
+            itemsByDimension: {
+                ou: DEFAULT_UI.itemsByDimension.ou,
```
Comment (edoardo) [inline packages/app/src/reducers/__tests__/ui.spec.js]: `[ouId]`?

File: packages/app/src/reducers/current.js [hunk 7]
```diff
@@ -12,17 +12,24 @@ export const DEFAULT_CURRENT = {};
...
-    columns: [createDimension(peId, [action.value.yearOnYearCategory])],
-    rows: [createDimension(dxId, [action.value.itemsByDimension.dx[0]])],
-    filters: getAxesFromUi(action.value).filters.filter(
-        f => f.dimension !== dxId
-    ),
-    yearlySeries: action.value.yearOnYearSeries,
-});
+const getYearOnYearCurrentFromUi = (state, action) => {
+    const dxItem = action.value.itemsByDimension.dx
+        ? action.value.itemsByDimension.dx[0]
```
Comment (edoardo) [inline packages/app/src/reducers/current.js]: `[dxId]`?

File: packages/app/src/reducers/ui.js [hunk 8]
```diff
@@ -27,16 +29,19 @@ export const SET_UI_YEAR_ON_YEAR_SERIES = 'SET_UI_YEAR_ON_YEAR_SERIES';
...
-        rows: ['pe'],
-        filters: ['ou'],
+        columns: [dxId],
+        rows: [peId],
+        filters: [ouId],
     },
     itemsByDimension: {
-        dx: [],
         pe: ['LAST_12_MONTHS'],
         ou: ['USER_ORGUNIT'],
```
Comment (edoardo) [inline packages/app/src/reducers/ui.js]: `[dxId]`, `[peId]`?

File: packages/app/src/reducers/__tests__/current.spec.js (hunk 3 above)
Comment (edoardo) [reply packages/app/src/reducers/__tests__/current.spec.js]: here I meant this: `[dxId]: ['dxItemId1', 'dxItemId2'],`

File: packages/app/src/components/Dimensions/DataDimension/DataDimension.js (hunk 2 above)
Comment (janhenrikoverland) [reply packages/app/src/components/Dimensions/DataDimension/DataDimension.js]: Fixed

File: packages/app/src/reducers/__tests__/ui.spec.js (hunk 5 above)
Comment (janhenrikoverland) [reply packages/app/src/reducers/__tests__/ui.spec.js]: Fixed

**Pull request merged**